components/nexus-core/src/main/java/org/sonatype/nexus/util/file/DirSupport.java

```diff
@@ -394,6 +399,45 @@ public static boolean copyIfExists(final Path from, final Path to, final @Nullab
   // MOVE: recursive copy of whole directory tree and then deleting it
 
   /**
+   * Perform a move of existing directory (empty or not, does not matter) on same FileStore (volume or partition).
+   */
+  private static void sameFileStoreMove(final Path from, final Path to) throws IOException {
+    Files.move(from, to, StandardCopyOption.REPLACE_EXISTING);
+  }
+
+  /**
+   * Perform a pseudo-move of existing directory (empty or not, does not matter) between different FileStores (volume or
+   * partition) using {@link #copyDeleteMove(Path, Path, Predicate)} method.
+   */
+  private static void crossFileStoreMove(final Path from, final Path to) throws IOException {
+    copyDeleteMove(from, to, null);
+  }
+
+  /**
+   * Return {@code true} if paths {@code from} and {@code to} are located on same FileStore (volume or
+   * partition). The {@code from} must exists, while {@code to} does not have to.
+   */
+  private static boolean areOnSameFileStore(final Path from, final Path to) {
+    try {
+      final FileStore fromStore = Files.getFileStore(from); // from must exist
+      Path toExistingParent = to.normalize(); // to usually does not exists, is about to be created as part of move
+      while (toExistingParent != null && !Files.exists(toExistingParent)) {
+        toExistingParent = toExistingParent.getParent();
+      }
+      if (toExistingParent != null) {
+        final FileStore toStore = Files.getFileStore(toExistingParent);
+        return fromStore.equals(toStore);
```

Comment:
Is it documented anywhere that this is how you detect different underlying filesystems?